src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -2166,7 +2166,7 @@ private static bool IsThisOrTypeOrNamespace(MemberAccessExpressionSyntax memberA
 
             var initializerType = semanticModel.GetTypeInfo(equalsValueClause.Value, cancellationToken).Type;
 
-            if (!type.Equals(initializerType))
+            if (!Equals(type, initializerType))
```

Comment:
We should bail out and return true if either `type` or `initializerType` is non-null. @Pasick I presume your analyzer flags this because there was a null check done on type earlier?